### PR TITLE
Docs for new range and autorange options + update Plotly.js version

### PR DESCRIPTION
**Pull request by LiamConnors**
@archmoj 

### Documentation PR

- [x] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [x] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [x] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [x] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [x] Every new/modified example is independently runnable
- [x] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [x] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [x] The random seed is set if using randomly-generated data in new/modified examples
- [x] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [x] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [x] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [x] Data frames are always called `df`
- [x] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [x] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [x] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [x] `fig.show()` is at the end of each new/modified example
- [x] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [x] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

### CODE REVIEW

File: doc/python/axes.md [hunk 1]
```diff
@@ -724,6 +728,21 @@ fig.update_yaxes(range=[9, 3])
...
+
+To use a reversed axis while specifying only an upper or lower bound for the range, set `autorange="reversed"`:
+
+```python
+import plotly.express as px
+
+df = px.data.iris()
+
+fig = px.scatter(df, x="sepal_width", y="sepal_length", facet_col="species")
+fig.update_yaxes(range=[9, None], autorange="reversed")
```
Comment (archmoj) [inline doc/python/axes.md:741]:
```suggestion
fig.update_yaxes(range=[9, None], autorange="min reversed")
```

File: doc/python/axes.md [hunk 2]
```diff
@@ -724,6 +728,21 @@ fig.update_yaxes(range=[9, 3])
 fig.show()
 ```
 
+*New in 5.16*
+
+To use a reversed axis while specifying only an upper or lower bound for the range, set `autorange="reversed"`:
```
[the remark this answers is not in the document]
Comment (LiamConnors) [reply doc/python/axes.md:733]: Thanks @archmoj
So I think I need to edit the text too:

```
*New in 5.17*

To use a reversed axis while specifying only a lower bound for the range, set `autorange="min reversed"`:
```
Does it count as specifying the lower bound if this is how it renders? Or are you technically setting the upper bound when you reverse the axis. 

image

[the remark this answers is not in the document]
Comment (archmoj) [reply doc/python/axes.md]: @Liam that's a good explanation.
Here is demo showing `min reversed` and second range being `null`: [codepen](https://codepen.io/MojtabaSamimi/pen/jOXBNdQ?editors=0010).
And this [codepen](https://codepen.io/MojtabaSamimi/pen/vYvxBMZ?editors=0010) iluustrates `max reversed` and first range being `null`.

**Review by alexcjohnson — APPROVED** [review]
💃

**Pull request merged**